Makefile

```diff
@@ -1,36 +1,41 @@
 
 # Default configuration
-SHELL     := /bin/bash
-PYTHON    := python3
-PIP       := ${PYTHON} -m pip
-LINTER    := pycodestyle
-FORMATTER := black
-MAKE_TAG  := --no-print-directory
+SHELL       := /bin/bash
+PYTHON      := python3
+PIP         := ${PYTHON} -m pip
+LINTER      := pycodestyle
+FORMATTER   := black
+MAKE_TAG    := --no-print-directory
 
 # Virtual env
-VENV_NAME = venv
+VENV_NAME   := venv
+VENV_PYTHON := ${VENV_NAME}/bin/${PYTHON}
 
 
 .PHONY: echo-cyan, echo-purple, echo-green, requirements, lint, format, clean
 
 
 ## Create virtualenv
 venv:
-	@$(MAKE) $(MAKE_TAG) echo-cyan msg="Creating and seting up a virtualenv..."
+	@$(MAKE) $(MAKE_TAG) echo-cyan msg="Creating and setting up a virtualenv..."
 	@$(PYTHON) -m venv $(VENV_NAME)
-	@. $(VENV_NAME)/bin/activate; \
-	$(PIP) install --upgrade pip; \
-	$(PIP) install -r requirements.txt
-	@$(MAKE) $(MAKE_TAG) echo-green msg="Virtual created environment successfully ! ✨"
-	@echo
-	@echo "To activate it, please run 'source venv/bin/activate'"
+	@. $(VENV_NAME)/bin/activate;
```

Comment:
All of this was intentional. You shouldn't make these changes.

After sourcing the venv using `. venv/bin/activate`, you need to run the next commands in the environment. like that `. venv/bin/activate; another_command; another_command_again; etc`

This way, the commands run in the activated environment. If you don't, I'm afraid it doesn't install it in the environment... It was an issue I found in stackoverflow

Also, we need to upgrade pip right after the env is created. That's necessary